components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/APIDescriptionGenUtil.java

```diff
@@ -36,35 +36,48 @@
+            if (requestPerMinute >= 1) {
+                String description = DESCRIPTION.replaceAll("\\[1\\]", Integer.toString(requestPerMinute));
+                return description;
+            }
+            return DESCRIPTION;
+        } catch (NullPointerException npe) {
+            String msg = "Policy could not be parsed correctly based on http://schemas.xmlsoap.org/ws/2004/09/policy " +
+                         "specification";
+            log.warn(msg);
+            throw new AppManagementException(msg);
+        }
+    }
+
+    public static int generateMaxCountFromPolicy(OMElement policy) throws
+            AppManagementException {
         OMElement maxCount = null;
         OMElement timeUnit = null;
         int requestPerMinute;
         try {
             maxCount = policy.getFirstChildWithName(AppMConstants.POLICY_ELEMENT).getFirstChildWithName
-                    (AppMConstants
-                             .THROTTLE_CONTROL_ELEMENT).getFirstChildWithName(AppMConstants.POLICY_ELEMENT).
+                    (AppMConstants.THROTTLE_CONTROL_ELEMENT).getFirstChildWithName(AppMConstants.POLICY_ELEMENT).
                     getFirstChildWithName(AppMConstants.THROTTLE_MAXIMUM_COUNT_ELEMENT);
             timeUnit = policy.getFirstChildWithName(AppMConstants.POLICY_ELEMENT).getFirstChildWithName
-                    (AppMConstants
-                             .THROTTLE_CONTROL_ELEMENT).getFirstChildWithName(AppMConstants.POLICY_ELEMENT).
+                    (AppMConstants.THROTTLE_CONTROL_ELEMENT).getFirstChildWithName(AppMConstants.POLICY_ELEMENT).
                     getFirstChildWithName(AppMConstants.THROTTLE_UNIT_TIME_ELEMENT);
             //Here we will assume time unit provided as milli second and do calculation to get requests per minute.
             if (maxCount.getText().isEmpty() || timeUnit.getText().isEmpty()) {
-                String msg = AppMConstants.THROTTLE_MAXIMUM_COUNT_ELEMENT.toString() + "or"
-                             + AppMConstants.THROTTLE_UNIT_TIME_ELEMENT.toString() + " element data found empty in " +
+                String msg = AppMConstants.THROTTLE_MAXIMUM_COUNT_ELEMENT.toString() + "or" +
+                             AppMConstants.THROTTLE_UNIT_TIME_ELEMENT.toString() + " element data found empty in " +
                              "the policy.";
                 log.warn(msg);
                 throw new AppManagementException(msg);
             }
-            requestPerMinute = (Integer.parseInt(maxCount.getText().trim()) * 60000) / (Integer.parseInt(timeUnit.getText().trim()));
-            if (requestPerMinute >= 1) {
-                String description = DESCRIPTION.replaceAll("\\[1\\]", Integer.toString(requestPerMinute));
-                return description;
-            }
-            return DESCRIPTION;
+            requestPerMinute = (Integer.parseInt(maxCount.getText().trim()) * 60000) /
+                               (Integer.parseInt(timeUnit.getText().trim()));
```

Comment:
That why we are having NullPointerException catch block. But as Dinusha commented we need to remove null catching and refactor those code with null checking. Will do it for the whole class and for the tracking purpose I raised a jira.